docs/sections/user_guide/yaml/components/make_hgrid.rst

```diff
@@ -0,0 +1,241 @@
+.. _make_hgrid_yaml:
+
+make_hgrid
+==========
+
+Structured YAML to run :ufs-utils:`make_hgrid<make-hgrid>` is validated by JSON Schema and requires the ``make_hgrid:`` block, described below. If ``make_hgrid`` is to be run via a batch system, the ``platform:`` block, described :ref:`here <platform_yaml>`, is also required.
+
+YAML keys and their descriptions are taken largely verbatim from the ``make_hgrid`` tool source code, available `here <https://github.com/NOAA-GFDL/FRE-NCtools/blob/master/tools/make_hgrid/make_hgrid.c>`_.
+
+Here is a prototype UW YAML ``make_hgrid:`` block, explained in detail below:
+
+.. highlight:: yaml
+.. literalinclude:: ../../../../shared/make_hgrid.yaml
+
+UW YAML for the ``make_hgrid:`` Block
+-------------------------------------
+
+execution:
+^^^^^^^^^^
+
+See :ref:`here <execution_yaml>` for details.
+
+run_dir:
+^^^^^^^^
+
+The path to the directory where ``make_hgrid`` will write its outputs.
+
+config:
+^^^^^^^
+
+Describes the required parameters to run a ``make_hgrid`` configuration. 
+
+
+grid_type:
+""""""""""
+
+``grid_type`` is the primary required key that defines the grid and additional required keys. It is specified with one of the following recognized key words:
```

Comment:
It's so awful that I hesitate to mention it, but: Our schema currently ensures that, when `grid_type` is set to some value, that certain other `config` items are defined as well. But it does not, I think, ensure that _other_ values are unset -- and, if they are, I think they'll get translated into `--cli_flags` too, possibly resulting in invalid run commands. Maybe not: Maybe the `make_hgrid` program is resilient against this and ignores irrelevant flags. Hopefully! It might be possible to model this stricter use case with `"additionalProperties": false` lines in the `"then"` clauses we already have, or perhaps using a `oneOf` clause instead of `allOf`, to make sure that exactly one valid set of arguments is specified. It might be interesting to do a quick manual check to see how `make_hgrid` deal with an extra, inapplicable argument, so set our minds (or at least mine) at ease in case it behaves graceully. Otherwise, we could revisit this topic if/when a user runs into difficulty.

Reply:
To make this more unnerving, its also not especially clear which parameters are not "required" but are still necessary anyway... unfortunately, it may be up to the users to give us more examples

Reply:
I've parsed it to pseudocode and added that to my summary doc, working on converting that to even more conditionals

Reply:
Saving further details for future work